src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -438,10 +439,42 @@ public override BoundNode VisitObjectInitializerMember(BoundObjectInitializerMem
 
         public override BoundNode VisitCall(BoundCall node)
         {
-            VisitCall(node.Method, null, node.Arguments, node.ArgumentRefKindsOpt, node.ArgumentNamesOpt, node.DefaultArguments, node);
-            CheckReceiverIfField(node.ReceiverOpt);
-            CheckReferenceToMethodIfLocalFunction(node, node.Method);
-            return base.VisitCall(node);
+            if (node.ReceiverOpt is BoundCall receiver1)
+            {
+                var calls = ArrayBuilder<BoundCall>.GetInstance();
+
+                calls.Push(node);
+                node = receiver1;
+
+                while (node.ReceiverOpt is BoundCall receiver2)
+                {
+                    calls.Push(node);
+                    node = receiver2;
+                }
+
+                CheckReceiverIfField(node.ReceiverOpt);
+                this.Visit(node.ReceiverOpt);
```

Comment:
The ordering for these two calls (`CheckReceiverIfField(node.ReceiverOpt)` and `this.Visit(node.ReceiverOpt)`) seems a bit different than in the original code. I'm not sure if it matters...
Is that intentional?

In comparison, the order for single invocation case below matches the previous order:
1. VisitCall
2. CheckReceiverIfField
3. CheckReferenceToMethodIfLocalFunction
4. Visit receiver
5. Visit arguments

Reply:
As far as I understand there is no order dependency between  Visit and Check methods.